### PR TITLE
Js interview Кучин Илья 

### PR DESCRIPTION
**Pull request by hedinant**

### CODE REVIEW

File: List.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+ constructor() {
+   this.state = {
+     data: null,
+     isDataFetched: false,
+     height: this.props.height
+   };
+ }
+ 
+ render() {
+   if (!this.state.isDataFetched) {
```
Comment (ilyakuchin) [inline List.js:29]: Move data fetch to componentDidMount

File: List.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+             {item.value}
+           </div>
+         );
+       })}
+     </div>
+   );
+ }
+ 
+ componentWillMount() {
+   this.props.parentRef.addEventListener('scroll', this.handleParentContainerScroll);
```
Comment (ilyakuchin) [inline List.js:59]: Event listeners shouldn't be added after component unmounting 
Upd: componentWillMount is deprecated. Use componentDidMount or constructor instead

File: List.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+* Высота списка задаётся через свойство из props, но если случился хотя бы один клик на элементе списка,
+* то высота должна стать 'auto'.
+* Также список должен отслеживать скролл на родительском контейнере (parentRef) и через коллбек
+* отправлять свои координаты.
+*/
+class List extends PureComponent {
+ ref = createRef();
+ 
+ shouldComponentUpdate(nextProps, nextState) { 
+   return nextState.height !== this.state.height;
```
Comment (ilyakuchin) [inline List.js:17]: Probably === instead of !==

File: List.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+     this.setState({
+       isDataFetched: true
+     });
+   }
+ 
+   return (
+     <div height={this.state.height}>
+       {this.state.data.map((item) => {
+         return (
+           <div onClick={this.handleItemClick(item.id)}>
```
Comment (ilyakuchin) [inline List.js:49]: add key attribute

Comment (hedinant) [reply List.js:49]: В обсуждении.

**Pull request closed without merging**